packages/app/src/components/approve/ApproveToken.tsx

```diff
@@ -0,0 +1,64 @@
+import { FC, MouseEventHandler } from "react";
+import { Button, Container, Tooltip, OverlayTrigger, Spinner } from "react-bootstrap";
+import { constants, Contract } from "ethers";
+import { addresses } from "@tender/contracts";
+import { useContractFunction } from "@usedapp/core";
+
+type Props = {
+  protocolName: string;
+  tokenSymbol: string;
+  isTokenApproved: boolean;
+  tokenAddress: Contract;
+};
+
+const ApproveToken: FC<Props> = ({ tokenSymbol, protocolName, isTokenApproved, tokenAddress }) => {
+  const { state: approveTx, send: approveToken } = useContractFunction(tokenAddress, "approve");
+
+  const handleApproval: MouseEventHandler<HTMLDivElement> = async (e) => {
+    e.preventDefault();
+
+    if (!isTokenApproved) {
+      await approveToken(addresses[protocolName].swap, constants.MaxUint256);
```

Comment:
If we add the spender as a prop rather than using `addresses[protocolName].swap` here we can use approval for any token towards any contract (see my previous comment)